### ISSUE TITLE
Unclear how to run only a single rule

### ISSUE THREAD

**Issue by egilll** [coderaiser/putout#77]
I'm unable to figure out how to run Putout with only a single rule enabled. One would assume that `putout src --enable remove-unused-variables`, or that only having one `plugin`/`rule` specified in .putout.json would do the trick, but that turns out not to be the case, instead all default rules are enabled.

(This is not an issue as such, but if anyone could tell me how to do this I would like to add it to the docs. I do see now what `--enable` does, but it's not quite what one would expect.)

**Comment by coderaiser** [comment]
The thing is most rules is suggested to be used, so they enable by default. Anyways if you want to set a baseline, you can disable all enabled rules with:

```sh
putout src —disable-all
```
This command will create `.putout.json` with all found rules disabled.

Then next command do the thing

```sh
putout src --enable remove-unused-variables
```

You can add this to docs ;). 
Is it works for you?

**Comment by milahu** [comment]
> ```
> putout src --disable-all
> ```
>
> This command will create `.putout.json` with all found rules disabled.

not really.
this command will do a transform-dryrun on src, which can take loooong,
and then will write `.putout.json`, with only SOME rules disabled
-> will disable only the rules that match the source files

**Comment by coderaiser** [comment]
> error no plugin found for a rule: "<rulename>" parser

Do you mean this message?

```
 1:1  error   no plugin found for a rule: "add-return-await"  parser
```

`add-return-await` is part of `promises` since [putout v19](https://github.com/coderaiser/putout/releases/tag/v19.0.0).

Let's suppose we have a file `index.js`:


```js
const unused = 5;

module.exports = function() {
    return promise();
}

async function promise(a) {
    return Promise.reject(Error('x'));
}
```

`putout index.js --disable-all` will find next errors:

```sh
 1:4   error   "unused" is defined but never used                        remove-unused-variables
 7:23  error   "a" is defined but never used                             remove-unused-variables
 3:0   error   Arrow functions should be used                            convert-to-arrow-function
 1:0   error   "use strict" directive should be on top of commonjs file  strict-mode/add
 8:4   error   Reject is useless in async functions, use throw instead   promises/convert-reject-to-throw
 4:11  error   Async functions should be called using await              promises/add-missing-await
 7:0   error   Useless async should be avoided                           promises/remove-useless-async
```

And create config file `.putout.json` with:

```
{
    "rules": {
        "remove-unused-variables": "off",
        "convert-to-arrow-function": "off",
        "strict-mode/add": "off",
        "promises/convert-reject-to-throw": "off",
        "promises/add-missing-await": "off",
        "promises/remove-useless-async": "off"
    }
}
```

Then running `putout index.js --enable convert-commonjs-to-esm` will update config with:

```diff
{
    "rules": {
        "remove-unused-variables": "off",
        "convert-to-arrow-function": "off",
        "strict-mode/add": "off",
        "promises/convert-reject-to-throw": "off",
        "promises/add-missing-await": "off",
-       "promises/remove-useless-async": "off"
+       "promises/remove-useless-async": "off",
+       "convert-commonjs-to-esm": "on"
    }
}
```

Then `putout --fix index.js` will do the thing and update `index.js` with:

```js
const unused = 5;

export default function() {
    return promise();
};

async function promise(a) {
    return Promise.reject(Error('x'));
}
```

So @milahu you case with `src` will look this way:

```sh
putout src --disable-all && putout src --enable convert-commonjs-to-esm && putout src --fix
```

This command will disable all rules that `Putout` can find right now and enable single rule. All `Putout` rules made for good and highly suggested to be used, they all enabled in this and all my other repositories. Anyways you can always disable what you don't need.

**Comment by milahu** [comment]
aah! : )

my error was to combine `--fix` and `--disable-all` in one call

in that case, putout should throw an error like "only one command per call"
where commands are `--fix`, `--disable-all`, `--enable`, ...

**Comment by coderaiser** [comment]
Add ability to show error messages when both options used in [putout v19.3.0](https://github.com/coderaiser/putout/releases/tag/v19.3.0) 🎉 .

![image](https://user-images.githubusercontent.com/1573141/131705225-f91dce79-5776-46db-90e5-3a0485452901.png)

**Comment by egilll** [comment]
Ah I see, thank you. 

I think it would be a pretty useful feature to be able to run a single rule without setting up a .putout.json file, it would make one-off uses easier and would make it easier to try out specific transformations.

Would you be interested in me sending in a pull request that would add this type of granularity to the command line interface, as in, allowing the use of the CLI without the .putout.json file?

**Comment by coderaiser** [comment]
We can try to do this, but first would be amazing if you tell me how do you see this? Is it a new `cli-option`? How should it look from user perspective?

**Comment by coderaiser** [comment]
A couple things to remember:
- every rule is a `plugin`, so it should be mentioned;
- `formatter` should be defined;
- some directories should be `ignored`;

Rich command line options and lack of config files is one of a problems `browserify` had, and as opposite one of a reasons `webpack` became so popular.

On the long run configuration file is very useful, and all rules made for good and suggested to be used by everybody in the same way as `ESLint` rules used.

**Comment by egilll** [comment]
Hm yes I do see your point.

My idea was to propose something along the lines of:

```sh
putout
  --rules apply-optional-chaining,merge-if-statements
  --fix
```

Or possibly:

```sh
putout
  --enable apply-optional-chaining
  --enable merge-if-statements
  --fix
```

But alternatively I do like Milahu's suggestion of disabling all defaults in the config:

```diff
{
+  "defaults": "off",
  "rules": {
    "apply-destructuring/array": "on"
  },
  "processors": [["javascript", "on"]]
}
```

as it would make Putout's behaviour more in line with one's expectations. For example, if one later adds some new code pattern one wouldn't have to think "Do I have to turn all the defaults off in Putout again?". It might also to make the config file more managable for those who are starting out with Putout and who are being cautious regarding transformations. At least in my case it wasn't feasible to start with all defaults enabled since there are a handful of rules that cause my code to break in some subtle way, therefore I've been turning them on one by one to be able to review the outcome.

**Comment by coderaiser** [comment]
@egilll this is a very good idea, especially:

```js
putout
  --enable apply-optional-chaining
  --enable merge-if-statements
  --fix
```

But we should add a couple more options to avoid reading `.putout.json` to set `plugins `and `formatter`, so it will look like this, I think:

```js
putout
  --no-config
  --enable apply-optional-chaining
  --enable merge-if-statements
  --format progress-bar
  --plugin apply-optional-chaining
  --plugin merge-if-statements 
  --fix
```
Also we should remember that one plugin can contain a bunch of rules, like [@putout/plugin-promises](https://github.com/coderaiser/putout/tree/master/packages/plugin-promises).

**Comment by coderaiser** [comment]
One more thing to remember is: each rule [does one thing and does it well](https://en.wikipedia.org/wiki/Single-responsibility_principle). For example [convert-commonjs-to-esm](https://github.com/coderaiser/putout/tree/v20.2.0/packages/plugin-convert-commonjs-to-esm) works better with newly created `ESLint` rule [no-unresolved](https://github.com/coderaiser/putout/tree/v20.2.0/packages/eslint-plugin-putout/lib/no-unresolved). And such rules as:
- [merge-duplicate-imports](https://github.com/coderaiser/putout/tree/v20.2.0/packages/plugin-merge-duplicate-imports);
- [merge-destructuring-properties](https://github.com/coderaiser/putout/tree/v20.2.0/packages/plugin-merge-destructuring-properties);
- [remove-nested-blocks](https://github.com/coderaiser/putout/tree/v20.2.0/packages/plugin-remove-nested-blocks)
- [declare-unedefined-variables](https://github.com/coderaiser/putout/tree/v20.2.0/packages/plugin-declare-undefined-variables)
- etc...

Helps a lot to avoid duplicating work, and makes creating new rule much easier, because you always know that you covered :).

**Comment by coderaiser** [comment]
I close this since we already have [🤷‍♂️ What is `Ruler`](https://github.com/coderaiser/putout#%EF%B8%8F-what-is-ruler) chapter in documentation of 🐊 Putout.

**Comment by clintonb** [comment]
This is quite confusing. Most tools of this type make rules opt-in. This one is opt-out, and we have to run the tool over the code base to know _what_ to opt-out of. @egilll's suggestion is how I expected things to work. It would be great if a future version adopts the opt-in behavior to be less confusing.

**Comment by coderaiser** [comment]
@clintonb, the thing is, when you go `ESLint` way, `recommended` is great, but when you need more you should search for a great amount of rules. To create [@putout/eslint-config](https://github.com/coderaiser/putout/tree/master/packages/eslint-config) I had to read information about all auto fixable `ESLint` rules. I don’t think most users will do that. Why? If you have [eslint-config-hardcore](https://github.com/EvgenyOrekhov/eslint-config-hardcore) which do all the things and much more.

So 🐊`Putout` not just regular linter, it instills the best development practices with ability to `disable` any rule and set `baseline` to start from. But it is a practical instrument without false positives, because it 🔍`finds` only what it can 🔧`fix`.

On the other hand no one forbids you from creating any kind of a wrapper, that uses API of 🐊`Putout` for it’s needs. It has no rules enabled, you starting from scratch, `tabula rasa`. A lot of effort invested to make the simplest `AST-transforming` framework in the world 😏.

Also I want mention that 🐊`Putout` is different from other tools. And it is it’s strong point. This is the reason why you will love it or hate it. It’s your choice, but you will definitely always have a strong reason for that.

**Comment by coderaiser** [comment]
Just landed 🐊`Putout` [v22.4.0](https://github.com/coderaiser/putout/releases/tag/v22.3.4) with ability to override default config with help of `PUTOUT_CONFIG_FILE` env variable:

```
PUTOUT_CONFIG_FILE=.putout-off.json putout .
```

With `.putout-off.json`:

```json
{
    "formatter": "progress-bar",
    "processors": [
        "javascript",
        "json",
        "markdown",
        "ignore",
        "yaml",
        "css"
    ],
    "ignore": [
        "**/node_modules",
    ]
}
```

**Comment by Oleg2tor** [comment]
> Just landed 🐊`Putout` [v22.4.0](https://github.com/coderaiser/putout/releases/tag/v22.4.0) with ability to override default config with help of `PUTOUT_CONFIG_FILE` env variable:
> 
> ```
> PUTOUT_CONFIG_FILE=.putout-off.json putout . --no-config
> ```
> 
> With `.putout-off.json`:
> 
> ```json
> {
>     "formatter": "progress-bar",
>     "rules": {
>         "remove-unused-variables": "on"
>     },
>     "processors": [
>         "javascript",
>         "json",
>         "markdown",
>         "ignore",
>         "yaml",
>         "css"
>     ],
>     "ignore": [
>         "**/node_modules",
>     ],
>     "plugins": [
>         "remove-unused-variables"
>     ]
> }
> ```
> 
> It runs one rule `remove-unused-variables`. So you can add any rules you want :).

When i run like this it still process some errors:
jest/no-mocks-import (eslint),
jest/valid-title (eslint)
@typescript-eslint/no-unused-vars (eslint)

How to disable entirely?
